WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationThemeAdapter.java

```diff
@@ -0,0 +1,138 @@
+package org.wordpress.android.ui.accounts.signup;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.model.ThemeModel;
+import org.wordpress.android.ui.prefs.AppPrefs;
+import org.wordpress.android.widgets.WPNetworkImageView;
+
+import java.util.List;
+
+public class SiteCreationThemeAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private static final int VIEW_TYPE_HEADER = 0;
+    private static final int VIEW_TYPE_ITEM = 1;
+
+    private boolean mIsLoading;
+    private List<ThemeModel> mThemes;
+    private SiteCreationListener mSiteCreationListener;
+
+    public static class HeaderViewHolder extends RecyclerView.ViewHolder {
+        public final View progress;
+        public final TextView label;
+
+        HeaderViewHolder(View itemView) {
+            super(itemView);
+            this.progress = itemView.findViewById(R.id.progress_container);
+            this.label = itemView.findViewById(R.id.progress_label);
+        }
+    }
+
+    public static class ThemeViewHolder extends RecyclerView.ViewHolder {
+        private final WPNetworkImageView imageView;
+        private final TextView nameView;
+
+        ThemeViewHolder(View view) {
+            super(view);
+            imageView = (WPNetworkImageView) view.findViewById(R.id.theme_grid_item_image);
+            nameView = (TextView) view.findViewById(R.id.theme_grid_item_name);
```

Comment:
We can remove the `(WPNetworkImageView)` and `(TextView)` casts here.  If casting is preferred, we should add them to the [`HeaderViewHolder` constructor](https://github.com/wordpress-mobile/WordPress-Android/pull/7028/files#diff-f8bafc5526de92e494efc9a4e1430354R30) for consistency.

Reply:
Arggh. I still keep the casts as I'm not sure yet if AS2.x behaves correctly without them. I took your second proposal and added the cast in `HeaderViewHolder`'s constructor with 7810239.